refactor(transform): Single dataset, move conflict check up to lib.Save

dustmop opened the pull request:
Instead of a pair of read/write datasets on the starlark ds object, have only a single dataset. For checking conflicts, have Apply track which components are modified by a transform script. Compare that result against the components provided to the save path, but do so in a much more obvious way, up in lib/

b5 commented on transform/apply.go:
```diff
@@ -53,7 +54,7 @@ func NewTransformer(appCtx context.Context, loader dsref.Loader, pub event.Publi
 
 // Apply applies the transform script to a target dataset
-func (t Transformer) Apply(
+func (t* Transformer) Apply(
```
gofmt plz. Type should own the star

b5 commented on transform/startf/exec_step.go:
```diff
@@ -24,20 +24,20 @@ var log = golog.Logger("startf")
+	bodyFile   qfs.File
+	eventsCh   chan event.Event
+	thread     *starlark.Thread
+	changeList map[string]bool
```
again, use a `map[string]struct{}`

b5 reviewed (approved):
negative diffstat! Some small nits & then looks good to me

dustmop commented:
Ok, now switched to using PathMap to get the components that exist in the incoming dataset. Switched the `changeSet` (previously changeList) to use full component names now, which removes the need for the `shortNameToComponent` function.
